[nitin19011] iP

nitin19011 opened the pull request.

nitin19011 committed:
Duke's name change to Socks; Added greeting & exit

Implemented initial skeleton for the Socks chatbot:
- Added greeting message

- Added exit message

nitin19011 committed:
Added ability for socks to echo user's message

sock takes in userinput and if it matches with "bye"(not cases
sensitive) it prompts exit message and ends the session, otherwise, it
echo's the user's input command back on the terminal.

nitin19011 committed:
Added sock's ability to add tasks & list them

nitin19011 committed:
Added Sock's ability to mark and unmark tasks.

Socks is also able to handle user input errors like
- invalid mark
- invalid unmark

nitin19011 committed:
Fixed Coding Standard issues

osy4477 commented on src/main/java/Socks.java:
```diff
@@ -0,0 +1,55 @@
+    public static void main(String[] args) {
+
+        // Start message
+        String logo =
```
since the logo is a constant, I suggest that it can be a private final variable

osy4477 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,47 @@
+            System.out.println("You have no tasks to do! HOORAY!");
+        } else {
+            System.out.println("Here are your tasks:");
+            for (int i = 1; i < tasks.size() + 1; i++) {
```
good job on using correct spaces in your for loop

osy4477 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public void listTasks() { // show the list method
+        if (tasks.isEmpty()) {
```
good job on adding the space between "if" and "("

osy4477 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,47 @@
+            }
+        }
+    }
+}
```
good job on using the correct indentation in your TaskList class

osy4477 reviewed (commented):
Overall coding standards looks good. 👍

nitin19011 committed:
added appropriate classes to handle relevant methods

-added todo class
-added deadline class
-added event class
-added task manager class
-refactor main program to appropriate methods

nitin19011 committed:
added helpful comments and fixed code quality

NigelYeoTW commented on src/main/java/Socks.java:
```diff
@@ -0,0 +1,77 @@
+
+        default:
+            System.out.println("Please enter a valid command.");
+        }
```
Any reason why the break was left out of the default block?

NigelYeoTW commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,36 @@
+            for (int i = 0; i < allTasks.size(); i++) {
+                System.out.println("    " + (i + 1) + "." + allTasks.get(i));
+            }
+        }
```
Perhaps guard clause could reduce nesting

NigelYeoTW commented on src/main/java/Socks.java:
```diff
@@ -0,0 +1,77 @@
+            break;
+
+        case ("unmark"):
+            manager.markUndone(Integer.parseInt(parts[1]) - 1); // pass the task number to be unmarked as (task number - 1)
```
Perhaps you can reduce the magic numbers here as well as in the rest of your code below

NigelYeoTW commented on src/main/java/Socks.java:
```diff
@@ -0,0 +1,77 @@
+
+        while (programRunning) {
+            String input = sc.nextLine();
+            performUserCommands(input, manager);
```
Perhaps the function can be singular since it performs only one command in the function

NigelYeoTW commented on src/main/java/Socks.java:
```diff
@@ -0,0 +1,77 @@
+    public static void performUserCommands(String input, TaskManager manager) {
+        String[] parts = input.split(" "); // split the user input
+        String command = parts[0].toLowerCase(); // store first word of user input to decide what action to execute
+        String rest = input.substring(command.length()).trim(); // store the remaining section of user input
```
Perhaps the naming of the variables could be more intuitive, same for the rest of the code below and in the other files

NigelYeoTW commented on src/main/java/Socks.java:
```diff
@@ -0,0 +1,77 @@
+        switch (command) {
+        case ("bye"):
+            printExitMessage();
+            programRunning = false; // terminate the while loops by making programRunning false
```
Perhaps you can make this variable sound more like a boolean? I notice this in other areas of your code as well.

NigelYeoTW reviewed (commented):
Overall lgtm, though there seems to be some minor coding standard and convention errors, keep it up 🔥

nitin19011 committed:
Level-5: Add SocksException custom exception

nitin19011 committed:
Level-5: Added exception messages to Socks

nitin19011 committed:
Merge branch 'branch-Level-5' for Level-5: Handle Errors

nitin19011 committed:
Use Exceptions to handle errors

nitin19011 committed:
Implemented delete method to socks

nitin19011 committed:
added task storage functionality to socks

nitin19011 committed:
Merge branch 'branch-Level-6'

nitin19011 committed:
Merge branch 'branch-Level-7'

nitin19011 committed:
Added more-oop requirement classes and refactored code

nitin19011 committed:
Added find feature to Socks

nitin19011 committed:
Added header comments for public classes and methods

nitin19011 committed:
Level-9 Branch addition

nitin19011 committed:
branch-A-JavaDoc [not done prior]

nitin19011 committed:
A-UserGuide: add User Guide

nitin19011 committed:
Updated User Guide

nitin19011 committed:
updated jar file

nitin19011 committed:
Merging PRs branch for IP progression

nitin19011 committed:
Merge pull request #1 from nitin19011/Merging-PRs

Merging PRs
